doc/man3/EVP_PKEY_type.md

```diff
@@ -0,0 +1,43 @@
+=pod
+
+=head1 NAME
+
+EVP_PKEY_type,
+EVP_PKEY_id,
+EVP_PKEY_base_id,
+EVP_PKEY_set_type,
+EVP_PKEY_set_type_str,
+EVP_PKEY_set_alias_type,
+- examine and manipulate type of public key
+
+=head1 SYNOPSIS
+
+ #include <openssl/evp.h>
+
+ int EVP_PKEY_type(int type);
+ int EVP_PKEY_id(const EVP_PKEY *pkey);
+ int EVP_PKEY_base_id(const EVP_PKEY *pkey);
+ int EVP_PKEY_set_type(EVP_PKEY *pkey, int type);
+ int EVP_PKEY_set_type_str(EVP_PKEY *pkey, const char *str, int len);
+ int EVP_PKEY_set_alias_type(EVP_PKEY *pkey, int type);
+
+=head1 DESCRIPTION
+
+EVP_PKEY_set_alias_type() allows modifying a EVP_PKEY to use a
+different set of algorithms than the default. This is currently used
+to support SM2 keys, which use an identical encoding to ECDSA.
```

Comment:
I think adding a "=head1 EXAMPLE" section might be helpful.

I do also wonder whether it would be useful to have an "overview" type man page for SM2, e.g. see the other overviews in doc/man7. Or possibly a combined overview page covering SM2/3/4. Anyway - just a thought, and for a separate PR.